### PR TITLE
Add RDS articles feature

### PR DESCRIPTION
**Pull request by thewilkybarkid**
https://github.com/elifesciences/issues/issues/5347

For https://github.com/elifesciences/journal/pull/1220.

### CODE REVIEW

**Comment by giorgiosironi** [comment]
I've had some surprises with the aging info banner and YAML serialization/deserialization to render `parameters.yml`, so trying it out locally.

**Comment by giorgiosironi** [comment]
Looks good, just adding some sample data:
```
[10:47:40][elife@dev--journal:/srv/journal]$ grep rds -A 2 parameters.yml 
    example_rds_articles:
        '16846': https://repro.elifesciences.org/example.html
    
    rds_articles:
        '16846': https://hub.stenci.la/elife/30274/main/
    
    feature_rds:
        true
```

**Comment by giorgiosironi** [comment]
(incidentally, this will also rebuild this PR with the pillars now available in `builder-configuration`)

File: salt/pillar/journal.sls [hunk 1]
```diff
@@ -68,6 +68,10 @@ journal:
 
     cache_control: private, no-cache, no-store, must-revalidate
 
+    feature_rds: true
+    # 16846 is in api-dummy
+    rds_articles:
+        '16846': https://hub.stenci.la/elife/30274/main/
```
Comment (thewilkybarkid) [inline salt/pillar/journal.sls]:
```suggestion
        '16846':
            display: https://hub.stenci.la/elife/30274/main/
```

Comment (giorgiosironi) [reply salt/pillar/journal.sls]: Oh yeah, missed that. I guess with the current `master` there's no test that could find out, thanks

**Comment by giorgiosironi** [comment]
Gets rendered to a weird indentation:
```
[10:56:05][elife@dev--journal:~]$ grep -A 2 rds /srv/journal/parameters.yml
    example_rds_articles:
        '16846': https://repro.elifesciences.org/example.html
    
    rds_articles:
        '16846':
          display: https://hub.stenci.la/elife/30274/main/
--
    feature_rds:
        true
```
Works?

**Comment by giorgiosironi** [comment]
Tested locally with the PR's branch, weird indentation works.

**Comment by thewilkybarkid** [comment]
YAML. 💪

**Pull request merged**